jsonschema-maven-plugin/src/main/java/com/github/victools/jsonschema/plugin/maven/SchemaGeneratorMojo.java

```diff
@@ -434,29 +466,46 @@ private void setModules(SchemaGeneratorConfigBuilder configBuilder) throws MojoE
      */
     private URLClassLoader getClassLoader() {
         if (this.classLoader == null) {
-            List<String> runtimeClasspathElements = null;
-            try {
-                runtimeClasspathElements = project.getRuntimeClasspathElements();
-            } catch (DependencyResolutionRequiredException e) {
-                this.getLog().error("Failed to resolve runtime classpath elements", e);
-            }
+            List<URL> urls = getUrls(Classpath.WITH_COMPILE_DEPENDENCIES);
```

Comment:
**suggestion (blocking)**: The `classpath` configuration parameter appears to be ignored.
```suggestion
            List<URL> urls = this.getUrls(this.classpath);
```

Reply:
No this is intentional. It is used when scanning for files using the reflections library. Here the classloader is retrieved and I thought it would be good, if it contained all dependencies. Then custom modules on the classpath can be loaded independent of the setting.